src/main/java/io/appform/memq/ActorSystem.java

```diff
@@ -0,0 +1,64 @@
+package io.appform.memq;
+
+import io.appform.memq.actor.Actor;
+import io.appform.memq.exceptionhandler.ExceptionHandlingFactory;
+import io.appform.memq.retry.RetryStrategyFactory;
+import lombok.val;
+
+import java.io.Closeable;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class ActorSystem implements Closeable {
+
+    private final ConcurrentHashMap<String, ExecutorService> executors;
+    private final ExecutorServiceProvider executorServiceProvider;
+    private final Map<String, ExecutorConfig> executorConfigMap;
+    private final List<Actor<?>> registeredActors;
+    private final RetryStrategyFactory retryStrategyFactory;
+    private final ExceptionHandlingFactory exceptionHandlingFactory;
+
+    public ActorSystem(ActorSystemConfig config,
+                       ExecutorServiceProvider executorServiceProvider) {
```

Comment:
Let people register their handlers. Delegate to default implementations in delegated constructors

Reply:
Done